modules/core/src/main/java/org/apache/ignite/internal/cluster/IgniteClusterImpl.java

```diff
@@ -324,8 +328,44 @@ public IgniteClusterImpl(GridKernalContext ctx) {
         }
     }
 
+    /** {@inheritDoc} */
+    @Override public ClusterState state() {
+        guard();
+
+        try {
+            return ctx.state().publicApiState(true);
+        }
+        finally {
+            unguard();
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public void state(ClusterState newState) throws IgniteException {
+        guard();
+
+        try {
+            if (newState == ACTIVE_READ_ONLY)
```

Comment:
no need verification here, it already exist in code near,
org.apache.ignite.internal.processors.cluster.GridClusterStateProcessor#changeGlobalState0

Reply:
I'd prefer to keep this check here.

Reply:
don`t support